dashboard/app/views/levels/_blockly.html.haml

```diff
@@ -205,13 +166,23 @@
           showInstructions();
         }
       },
-      hide_source: #{local[:hide_source]},
-      share: #{local[:share]},
-      no_padding: #{local[:no_padding]},
-      position: { blockYCoordinateInterval: 25 },
-      cdoSounds: CDOSounds
     };
-    $.extend(appOptions, #{app_options.to_json});
-    #{'console.log(appOptions)' unless Rails.env.production?};
-    #{app}Main(appOptions);
+    $.extend(appOptions, baseOptions);
+    // Turn string values into functions for keys that begin with 'fn_' (JSON can't contain function definitions)
+    // E.g. { fn_example: 'function () { return; }' } becomes { example: function () { return; } }
+    (function fixUpFunctions(node) {
+      if (typeof node !== 'object') return;
+      for (var i in node) {
+        if (/^fn_/.test(i)) {
+          try {
+            node[i.replace(/^fn_/, '')] = eval('(' + node[i] + ')');
+          } catch (e) { }
+        } else {
+          fixUpFunctions(node[i]);
+        }
+      }
+    })(appOptions.level);
```

Comment:
Yep that's the next step :)  I wanted to get an in-progress review first with a diff of the changes.  I'll extract the now-invariant JS to a file before merging.  Should the new file live in `/dashboard/apps/assets/javascripts` or `/apps/src`?